crypto/asymmetric.go

```diff
@@ -63,6 +66,21 @@ func encryptCurve25519(pubKey *libp2pc.Ed25519PublicKey, bytes []byte) ([]byte,
 	return ciphertext, nil
 }
 
+func publicToCurve25519(k *libp2pc.Ed25519PublicKey) (*[32]byte, error) {
```

Comment:
libp2p-crypto remove this `ToCurve25519` method and it's private key variant... but mentioned in the commit msg that can easily be pulled back in. Doing that here.